posthog/hogql_queries/query_runner.py

```diff
@@ -995,6 +995,13 @@ def get_cache_payload(self) -> dict:
             "version": 2,
         }
 
+        # add week_start_day to the payload for queries with weekly intervals
+        query_date_range = getattr(self, "query_date_range", None)
+        if query_date_range and query_date_range.interval_name == "week":
```

Comment:
I'm unsure if it's safe to assume that only queries with weekly grouping are affected. Queries like weekly active users or HogQL queries using `toStartOfWeek` could also fall into this. For simplicity I'd just include the setting into the payload.

I get that this is to avoid too many cache invalidations, but I think it'd be better for the long run if we just do that.